### PR TITLE
Grants frontend complete

### PR DESCRIPTION
**Pull request by varortz**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :cloud: :cloud: Description :cloud: :cloud:
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Revamped the `GrantCard` component to reflect the new fronted changes for grants
- Completed the frontend for the `GrantInfo` screen
- Added Icons for the `amount` and `deadline` fields
- Added a placeholder image for all grant cards and the `GrantInfo`
- Added a static button in the `GrantInfo` screen to view the full grant application in another browser 
- Added a static back button to exit a `GrantInfo` screen
- 4/23 UPDATE: redid some of the implementation to try and follow best practices. Issue -> Images for `GrantCard` component do not fully fill the desired width.
- Increased the opacity for the `GrantCard` component
- 4/30 UPDATE: added Linking support for the `View Application` button and functionality to the back arrow button on the `GrantInfo` page

### Screenshots :calling:
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @leexesther at the bottom."
image
image

## :zap: How to review :zap:
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- Navigate to the grants newsfeed by clicking on the grant icon in the navbar
- Scroll through the page and ensure everything looks consistent with the designs
- Click on individual grants to get redirected to a 'GrantInfo' screen
- Scroll through the 'GrantInfo' screen for a grant and ensure that everything is consistent with the designs

## :seedling: Next steps :seedling:
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Add navigation for the back button on the `GrantInfo` screen
- Add support for uploading images to backend & displaying them on the app
- Add functionality to the "View Application" button
- Include a share grant feature which would share the grant's application
- Implement a "new grant" notifier

### Online sources :information_source:
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs :raised_hands:
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @davidqing6432 @wangannie @leexesther

### CODE REVIEW

File: src/screens/GrantInfoScreen/GrantInfo.tsx [hunk 1]
```diff
@@ -1,19 +1,69 @@
...
+      </View>
+      <View style={styles.innerContainer}>
+        <View>
+          {/* might have to handle font size for longer titles */}
+          <Text style={styles.titleText}>{grantObj.title}</Text>
+        </View>
+        <View style={styles.imageContainer}>
+          <Image source={grantPlaceholder} style={styles.image} />
+        </View>
+        <RectButton
```
Comment (david-qing) [inline src/screens/GrantInfoScreen/GrantInfo.tsx:41]: rect button seems to automatically capitalize text, don't know if this should be a problem

File: src/screens/GrantInfoScreen/GrantInfo.tsx [hunk 2]
```diff
@@ -1,19 +1,69 @@
...
+            </Text>
+          </View>
+          <View style={styles.infoContainer2}>
+            <Text>
+              <Text style={{ fontWeight: 'bold' }}>Duration: </Text>
+              {grantObj.duration}
+            </Text>
+          </View>
+          <View style={styles.infoContainer2}>
+            <Text>
```
Comment (david-qing) [inline src/screens/GrantInfoScreen/GrantInfo.tsx]: extra wrapper, what is this for?

File: src/screens/GrantInfoScreen/GrantInfo.tsx [hunk 3]
```diff
@@ -1,19 +1,69 @@
...
+            </View>
+          </View>
+          <View style={styles.infoContainer2}>
+            <Text>
+              <Text style={{ fontWeight: 'bold' }}>Eligible Countries: </Text>
+              {grantObj.countries.join(', ')}
+            </Text>
+          </View>
+          <View style={styles.infoContainer2}>
+            <Text>
```
Comment (david-qing) [inline src/screens/GrantInfoScreen/GrantInfo.tsx]: extra wrapper

File: src/screens/GrantInfoScreen/GrantInfo.tsx (hunk 2 above)
Comment (varortz) [reply src/screens/GrantInfoScreen/GrantInfo.tsx]: The one with styles.infoContainer2? That is for the grant fields (eligible countries, duration, and brief/description) since they require different styling compared to the amount + deadline fields.

File: src/screens/GrantInfoScreen/GrantInfo.tsx (hunk 1 above)
Comment (varortz) [reply src/screens/GrantInfoScreen/GrantInfo.tsx:41]: Oh yeah you're right! I just used the RectButton that we've been using but I can make another button component if you want.

**Review by david-qing — APPROVED** [review]
looks good!
